btf/btf.go

```diff
@@ -698,24 +698,24 @@ func LoadSplitSpecFromReader(r io.ReaderAt, base *Spec) (*Spec, error) {
 
 // TypesIterator iterates over types of a given spec.
 type TypesIterator struct {
-	spec  *Spec
+	types []Type
 	index int
 	// The last visited type in the spec.
 	Type Type
 }
 
 // Iterate returns the types iterator.
 func (s *Spec) Iterate() *TypesIterator {
-	return &TypesIterator{spec: s, index: 0}
+	return &TypesIterator{types: s.types, index: 0}
```

Comment:
This will cause `TypesIterator.types` and `Spec.types` to share a backing array. Not currently a problem as we don't allow types to be deleted or reshuffled. If this is deliberate, a small comment confirming the decision would be useful.